core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -1166,6 +1166,8 @@ final class ShuffleBlockFetcherIterator(
       case ShuffleBlockBatchId(shuffleId, mapId, startReduceId, _) =>
         throw SparkCoreErrors.fetchFailedError(address, shuffleId, mapId, mapIndex, startReduceId,
           msg, e)
+      case ShuffleBlockChunkId(shuffleId, _, reduceId, _) =>
+        SparkCoreErrors.fetchFailedError(address, shuffleId, -1L, -1, reduceId, msg, e)
```

Comment:
This will propagate the `FetchFailure` with mapId/mapIndex = -1. Currently the driver will not unregister the mergeStatus because mapIndex = -1. Since the mergeStatus is still registered, the next attempt will still try to fetch the corrupted merged block.  Eventually the application will fail.
So, this requires a change in the driver as well. Also once we make that change, we should add a UT to `DAGScheduler` to verify that.

Reply:
Thanks for the detailed description. Will update accordingly then.